### ISSUE TITLE
optimize='dp' gets confused by singleton factors 

### ISSUE THREAD

**Issue by yaroslavvb** [dgasmith/opt_einsum#118]
This hangs with 100% utilization. 
```
import opt_einsum as oe
einsum_string = 'nlp,nlq->n'
views = oe.helpers.build_views(einsum_string, {'n': 30, 'i': 100, 'l': 100, 'p': 2, 'q': 2, 'z': 1})
path, path_info = oe.contract_path(einsum_string, *views,optimize='dp')
print(path_info)
```
Changing optimizer or increasing size of `z` factor works

**Comment by jcmgray** [comment]
Ah this is because of the costcapping strategy. I'll fix this in the upcoming tweaks to 'dp'.

**Comment by jcmgray** [comment]
This should be fixed by #119.
